temporal/src/androidMain/kotlin/BroadcastTemporalFlow.kt

```diff
@@ -0,0 +1,32 @@
+package com.juul.tuulbox.temporal
+
+import android.content.Intent
+import android.content.IntentFilter
+import com.juul.tuulbox.coroutines.flow.broadcastReceiverFlow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.conflate
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+
+@PublishedApi
+internal val temporalEventFilter = IntentFilter().apply {
+    addAction(Intent.ACTION_DATE_CHANGED)
+    addAction(Intent.ACTION_TIME_CHANGED)
+    addAction(Intent.ACTION_TIME_TICK)
+    addAction(Intent.ACTION_TIMEZONE_CHANGED)
+}
+
+internal actual inline fun <T> inlineTemporalFlow(
+    crossinline factory: () -> T
+): Flow<T> = flow {
+    emit(factory.invoke())
+    emitAll(broadcastReceiverFlow(temporalEventFilter).map { factory.invoke() })
```

Comment:
Would it be possible to output which Filter generated the event, as a parameter in the lambda?

Reply:
It wouldn't be hard to add an `Intent` argument to the `factory` lambda, but doing that would require the API not to match other platforms.